ci/build.yml

```diff
@@ -64,7 +64,7 @@ jobs:
     - template: build-unix.yml
       parameters:
         release_tag: $(release_tag)
-        name: $(name)
+        name: variables.name
```

Comment:
I noticed that `is_release` and `release_tag` are declared above under `variables`, maybe we need to do the same for `name`? or is `name` a magic variable provided by azure?

Reply:
It's a magic variable introduced by the `strategy` block - I think. If I really knew what's going on here we wouldn't be in this situation 🙂

Reply:
ah yes that sounds right, I just found the relevant part of the docs

> ```yaml
> strategy:
>   matrix: { string1: { string2: string3 } }
>   maxParallel: number
> ```
> 
> For each occurrence of string1 in the matrix, a copy of the job is generated. The name string1 is the copy's name and is appended to the name of the job. For each occurrence of string2, a variable called string2 with the value string3 is available to the job.
> https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/jobs-job-strategy?view=azure-pipelines#remarks

Reply:
So I think the crux of the issue is that we are trying to evaluate `parameters.name` in two different contexts, where the evaluation rules are different.

On the one hand, we want to inline it into some "strings", as in the parameter passed to `build.sh` and the name of the published artifacts. On the other hand, we want to expand it into "expressions", in various condition blocks. Those should be using different syntax (because why not), and the `template` mechanisme is a simple search-and-replace, which means maybe we need to pass the sam value in two different forms as two separate parameters, one for each use type. 🤔